storage/src/main/java/com/brandwatch/minibcr/storage/kafka/config/KafkaConsumerConfig.java

```diff
@@ -0,0 +1,44 @@
+package com.brandwatch.minibcr.storage.kafka.config;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.annotation.EnableKafka;
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.support.serializer.JsonDeserializer;
+
+@EnableKafka
+@Configuration
+public class KafkaConsumerConfig {
+
+    @Value("${spring.kafka.bootstrap-servers}")
+    private String bootstrapServers;
+
+    private static final String CONSUMER_GROUP_ID = "resource-matching";
+
+    @Bean
+    public ConsumerFactory<String, String> consumerFactory() {
+        Map<String, Object> props = new HashMap<>();
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, CONSUMER_GROUP_ID);
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        //TODO: Using JsonDeserializer below breaks the app, find a way to make it work
+        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
```

Comment:
From my understanding, using a JsonDeserializer would enable the consumer to recover the original object that was stored (in this case a Resource instance) instead of a String version of it. For some reason, if I try to use it the app breaks.

Reply:
Some of the errors are :
`Caused by: org.apache.kafka.common.errors.SerializationException: Error deserializing key/value for partition resources-0 at offset 13. If needed, please seek past the record to continue consumption.`

`Caused by: org.apache.kafka.common.errors.SerializationException: Can't deserialize data [[123, 34, 114, 101, 115, 111, 117, 114, 99, 101, 73, 100, 34, 58, 34, 49, 50, 50, 52, 48, 52, 49, 65,...]] from topic [resources]`


`Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of com.brandwatch.minibcr.common.domain.Resource (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)`